server/controllers/contest.js

```diff
@@ -0,0 +1,90 @@
+const asyncHandler = require("express-async-handler");
+const Contest = require('../models/Contest')
+
+// handler for creating a new contest object
+exports.createContest = asyncHandler(async (req, res, next) => {
+    const { title, description, user, prizeAmount, deadline } = req.body;
+
+    const contest = await Contest.create({ title, description, prizeAmount, deadline });
+
+    if (contest) {
+        res.status(201).json({
+            success: {
+                contest: contest
+            }
+        });
+    } else {
+        res.status(400);
+        throw new Error("invalid contest data");
+    }
+
+    next()
+
+});
+
+// handler for updating a contest object
+exports.updateContest = asyncHandler(async (req, res, next) => {
+
+    const id = req.params.id
+    const { description, prizeAmount, title } = req.body;
+
+    try {
+        const updated = await Contest.updateOne({ id: id }, { title: title, description: description, prizeAmount: prizeAmount });
+
+        if (updated.nModified > 0) {
+            res.status(200).json({
+                success: {
+                    contest: {
+                        title: title,
+                        description: description,
+                        prizeAmount: prizeAmount,
+                    }
+                }
+            });
+        } else {
+            throw Error("failed to updates the contest");
+        }
+
+    } catch (error) {
+        res.status = 500;
```

Comment:
status should be a function here, please update accordingly.

Reply:
done